core/services/webhook/validate.go

```diff
@@ -30,16 +36,28 @@ func ValidatedWebhookSpec(tomlString string, externalInitiatorManager ExternalIn
 		return jb, errors.Errorf("the only supported schema version is currently 1, got %v", jb.SchemaVersion)
 	}
 
-	var spec job.WebhookSpec
-	err = tree.Unmarshal(&spec)
+	var tomlSpec TOMLWebhookSpec
+	err = tree.Unmarshal(&tomlSpec)
 	if err != nil {
 		return jb, err
 	}
-	jb.WebhookSpec = &spec
-	if !spec.ExternalInitiatorName.IsZero() {
-		if _, err := externalInitiatorManager.FindExternalInitiatorByName(spec.ExternalInitiatorName.String); err != nil {
-			return jb, err
+	var externalInitiatorWebhookSpecs []job.ExternalInitiatorWebhookSpec
+	for _, eiSpec := range tomlSpec.ExternalInitiators {
+		ei, err := externalInitiatorManager.FindExternalInitiatorByName(eiSpec.Name)
```

Comment:
could pull this into a single query pretty easily

Reply:
Yup, I considered this. Decided not to for a couple of reasons:

1. These lookups are only ever called once (on creation time) so performance doesn't matter too much
2. This way used a minimum of new code

What I will do is use a multierr instead so that all missing EIs are reported in one go.